src/DssProxyActionsCharter.sol

```diff
@@ -0,0 +1,686 @@
+    }
+
+    function freeETH(
+        address charter,
+        address ethJoin,
+        uint256 wad
+    ) external {
+        // Unlocks WETH amount from the CDP
+        frob(charter, GemJoinLike(ethJoin).ilk(), -_toInt256(wad), 0);
+        // Exits WETH amount to proxy address as a token
+        CharterLike(charter).exit(ethJoin, address(this), wad);
+        // Converts WETH to ETH
+        GemJoinLike(ethJoin).gem().withdraw(wad);
+        // Sends ETH back to the user's wallet
+        msg.sender.transfer(wad);
+    }
+
+    function freeGem(
+        address charter,
+        address gemJoin,
+        uint256 amt
+    ) external {
+        // Unlocks token amount from the CDP
+        frob(charter, GemJoinLike(gemJoin).ilk(), -_toInt256(_convertTo18(gemJoin, amt)), 0);
+        // Exits token amount to the user's wallet as a token
+        CharterLike(charter).exit(gemJoin, msg.sender, amt);
+    }
+
+    function exitETH(
+        address charter,
+        address ethJoin,
+        uint256 wad
+    ) external {
+        // Exits WETH amount to proxy address as a token
+        CharterLike(charter).exit(ethJoin, address(this), wad);
+        // Converts WETH to ETH
+        GemJoinLike(ethJoin).gem().withdraw(wad);
+        // Sends ETH back to the user's wallet
+        msg.sender.transfer(wad);
+    }
+
+    function exitGem(
+        address charter,
+        address gemJoin,
+        uint256 amt
+    ) external {
+        // Exits token amount to the user's wallet as a token
+        CharterLike(charter).exit(gemJoin, msg.sender, amt);
+    }
+
+    function draw(
+        address charter,
+        bytes32 ilk,
+        address jug,
+        address daiJoin,
+        uint256 wad
+    ) external {
+        address vat = CharterLike(charter).vat();
+
+        // Generates debt in the CDP
+        int256 dart = _getDrawDart(charter, vat, jug, ilk, wad);
+        frob(charter, ilk, 0, dart);
+        // Allows adapter to access to proxy's DAI balance in the vat
+        if (VatLike(vat).can(address(this), address(daiJoin)) == 0) {
+            VatLike(vat).hope(daiJoin);
+        }
+        // Exits DAI to the user's wallet as a token
+        DaiJoinLike(daiJoin).exit(msg.sender, wad);
+    }
+
+    function wipe(
+        address charter,
+        bytes32 ilk,
+        address daiJoin,
+        uint256 wad
+    ) external {
+        address vat = CharterLike(charter).vat();
+        address urn = CharterLike(charter).getOrCreateProxy(address(this));
+
+        // Joins DAI amount into the vat
+        daiJoin_join(daiJoin, wad);
+        // Allows charter to access to proxy's DAI balance in the vat
+        if (VatLike(vat).can(address(this), address(charter)) == 0) {
+            VatLike(vat).hope(charter);
+        }
+        // Paybacks debt to the CDP
+        frob(charter, ilk, 0, _getWipeDart(vat, VatLike(vat).dai(address(this)), urn, ilk));
```

Comment:
```suggestion
        address vat = CharterLike(charter).vat();

        // Joins DAI amount into the vat
        daiJoin_join(daiJoin, wad);
        // Allows charter to access to proxy's DAI balance in the vat
        if (VatLike(vat).can(address(this), address(charter)) == 0) {
            VatLike(vat).hope(charter);
        }
        // Paybacks debt to the CDP
        frob(
            charter,
            ilk,
            0,
            _getWipeDart(
                vat,
                VatLike(vat).dai(address(this)),
                CharterLike(charter).getOrCreateProxy(address(this)),
                ilk
            )
        );
```

Reply:
A part from the variable removal, the `hope/nope` thing might be an extra safety measure before we can analyse better the potential of leaving a long term DAI allowance to the charter. I think today won't be a problem as the `frob` function is quite restrictive but might be more flexible in the future. So I think is better just to turn off the allowance at end.